tests/test_ratmama.py

```diff
@@ -0,0 +1,255 @@
+        # and assure it recognized it as a reconnect
+        assert async_callable_fx.was_called_with(
+            f"Ajdacho has reconnected! Case #{index}"
+        )
+
+    async def test_reconnect_with_changes(self,
+                                          async_callable_fx: AsyncCallableMock,
+                                          context_channel_fx: Context,
+                                          monkeypatch
+                                          ):
+        """
+        Tests the recognition of changes and the emission of the associated message.
+        Tests a single case only
+        """
+        # set up all our tracking
+        monkeypatch.setattr(context_channel_fx, 'reply', async_callable_fx)
+        rat_board = context_channel_fx.bot.board
+
+        # set our first message
+        monkeypatch.setattr(context_channel_fx, '_words_eol',
+                            ["Incoming Client: Ajdacho - System: Alrai - "
+                             "Platform: PC - O2: OK - Language: Polish (pl-PL)"
+                             ]
+                            )
+
+        # set the nickname of the announcer
+        monkeypatch.setattr(context_channel_fx._user, '_nickname', "RatMama[BOT]")
+        # and have it processed
+        await RatMama.handle_ratmama_announcement(context_channel_fx)
+        # remember the index (is important later!)
+        index = rat_board.find_by_name("Ajdacho").board_index
+
+        # prepare second announcement, this one has different details,
+        # but is from the same commander
+        monkeypatch.setattr(context_channel_fx, '_words_eol',
+                            ["Incoming Client: Ajdacho - System: H - Platform: XB - "
+                             "O2: NOT OK - Language: Polish (pl-PL)"
+                             ]
+                            )
+
+        # and handle it
+        await RatMama.handle_ratmama_announcement(context_channel_fx)
+        # make sure it recognized it as a reconnect
+        assert async_callable_fx.was_called_with(
+            f"Ajdacho has reconnected! Case #{index}"
+        )
+
+        # as well as the fact some stuff changes
+        assert async_callable_fx.was_called_with(
+            "System changed! Platform changed! O2 Status changed, it is now CODE RED!"
+        )
+
+    async def test_no_action_on_wrong_nick(self, async_callable_fx: AsyncCallableMock,
+                                           context_channel_fx: Context,
+                                           monkeypatch):
+        """
+        Tests, that a wrong nickname has no associated action when handed to the RatMama handler.
+        """
+        # use a valid announcement
+        monkeypatch.setattr(context_channel_fx, '_words_eol',
+                            ["Incoming Client: Ajdacho - System: H - Platform: XB - "
+                             "O2: OK - Language: Polish (pl-PL)"
+                             ]
+                            )
```

Comment:
This monkeypatch sets `words_eol` to a strictly invalid value. 

`words_eol` is essentially generated by using `str.split(maxsplit=1)` repeatidly as to decompose the string into a list of strings. Consequently, a input that yields `words_eol[0]` with spaces in it necessitates other elements in this list.


That said, since its a test and you are up against the deadline I will differ this to a cleanup task.